### PR TITLE
⚡️ Improve melos and workflow setup

### PR DESCRIPTION
**Pull request by AlexV525**
- Reduce separate workflows and matrices to the minimum (`30+` → `3`).
- Add a script to apply file patches for corresponding SDKs.
- Add a script to produce `.melos_ignored_packages` for steps.
- Annotate test platforms for plugins.
- Fix linter recognition between Dart SDK versions.

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### Additional context and info (if any)

Melos has poor support around versions and packages filtering. It has `MELOS_PACKAGES` and `packageFilter` for sessions, but those fields are not working well across its major versions.

The main idea about filtering our packages is to prevent unsatisfied packages from being handled by melos process. For example, if `native_dio_adapter` requires Flutter `3.13.0` and the current Flutter is `2.15.1`, exclude it from packages.

To do this, I've added a script to serialize the melos workspace first, then determine whether the Dart version satisfies the package version. It produces package lists according to demand (satisfied packages or ignored packages).

### CODE REVIEW

**Comment by AlexV525** [comment]
`native_dio_adapter` has a different SDK requirement than others, so the `min` workflow fails. I didn't figure out a proper way to handle this with melos at the moment. Leaving this until I finish my exam. Any ideas are welcomed.

File: scripts/sdk_satisfaction
Comment (AlexV525) [inline scripts/sdk_satisfaction:1]: Compiled by Dart....Huge one

**Comment by AlexV525** [comment]
Melos. :)

**Comment by AlexV525** [comment]
See the edited top comment for more info.

**Comment by AlexV525** [comment]
Also cc @CaiJingLong to see if you have any idea regarding infrastructure abstraction abilities.

**Comment by kuhnroyal** [comment]
Personally, I don't like the combination of format/analyze/tests as this makes it harder to determine why the build failed, especially for new contributors.
Are there any CI limits reached regarding execution time or matrixes?

**Comment by AlexV525** [comment]
> Personally, I don't like the combination of format/analyze/tests as this makes it harder to determine why the build failed, especially for new contributors.

Good point. If we break it into steps, would it be good enough too?

> Are there any CI limits reached regarding execution time or matrixes?

As far as I know there are no limitations for public repos.

**Comment by AlexV525** [comment]
https://github.com/cfug/dio/actions/runs/6785748920/job/18444745192

Steps were split in the job. Now you're able to identify every verify step in a single workflow.

The core is time trade with speed. If we still use job strategies, it will cost massive redundant time since the workspace needs to be reconfigured each time a job starts. But it can display every action as a workflow.

**Review by CaiJingLong — APPROVED** [review]
LGTM

**Pull request merged**